### PR TITLE
feat: update auth in leaderboard to new style of documentation

### PR DESCRIPTION
**Pull request by rishtigupta**
## PR Description:
This commit refactors
-  The main and auth page for leaderboards
- The auth page for cache and topics (which is same contents as auth page for leaderboards)

## Questions:
- Do we need auth api ref in leaderboard documentation?
- Given our example codes are huge, they don't really fit well in the side column as implemented in this pr. Ideas/suggestions?

## Issue:
addresses a part of https://github.com/momentohq/dev-eco-issue-tracker/issues/957

### CODE REVIEW

File: docs/leaderboards/api-reference/auth.md [hunk 1]
```diff
@@ -16,74 +16,178 @@ The auth APIs create and manage API keys and tokens for Momento services. These
...
+---
+
+#### Returns
+---
 
 * Success
+
   - `apiKey`: string - the new auth token
-  - `refreshToken`: string - a refresh token that can be used with the [RefreshApiKey API](#refreshapikey-api) to refresh the token before it expires
+  - `refreshToken`: string - a refresh token that can be used with the [RefreshApiKey API](#refreshapikey-api) to refresh a token before it expires
```
Comment (anitarua) [inline docs/leaderboards/api-reference/auth.md]:
```suggestion
  - `refreshToken`: string - a refresh token that can be used with the [RefreshApiKey API](#refreshapikey) to refresh a token before it expires
```

File: docs/leaderboards/api-reference/auth.md [hunk 2]
```diff
@@ -16,74 +16,178 @@ The auth APIs create and manage API keys and tokens for Momento services. These
...
+
+#### Optional Parameters
+---
+
+#### Returns
 
 * Success
+
   - `authToken`: string - the new auth token
   - `refreshToken`: string - a refresh token that can be used with the [RefreshApiKey API](#refreshapikey-api) to refresh a token before it expires
```
Comment (anitarua) [inline docs/leaderboards/api-reference/auth.md]:
```suggestion
  - `refreshToken`: string - a refresh token that can be used with the [RefreshApiKey API](#refreshapikey) to refresh a token before it expires
```

**Review by anitarua — COMMENTED** [review]
overall I think it looks good! 

I like the wider code snippet width on this page. Could you update it for the main api ref page too?

Found some broken links, should be easy fix.

As for whether auth is needed here, maybe not, but this page could be directly copied to the cache and topics sections I think.

File: docs/leaderboards/api-reference/auth.md (hunk 1 above)
Comment (anitarua) [reply docs/leaderboards/api-reference/auth.md]: broken link

**Review by anitarua — DISMISSED** [review]
💚💚💚

**Review by anitarua — APPROVED** [review]

**Review by cprice404 — APPROVED** [review]

**Pull request merged**